app/Models/Concerns/HasFilters.php

```diff
@@ -117,7 +120,7 @@ public function scopeFilter(Builder $query, array $params = []): Builder
                     $this->filterDateRange($query, $key, $params);
                 } elseif ($key == 'races') {
                     $this->filterRaces($query, $value);
-                } elseif ($key == 'location_id') {
+                } elseif ($key == 'location_id' && $this->filterOption('children')) {
```

Comment:
We don't need the $this->filterOption here, we can make a fallback in the filterLocation function instead. Makes the code easier to read